src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,270 @@
+                    String[] inputArgs = input.split(" \\| ");
+                    addFileDataToList(inputArgs);
+                }
+            }
+        } catch (IOException e) {
+            System.out.print("\t(!) IOException error: get existing file data.\n" + breakLine());
+        } catch (IllegalStateException e) {
+            System.out.print("\t(!) IllegalStateException: (!) Invalid file contents.\n" + breakLine());
+        }
+        System.out.print("\tLoading Done :)\n" + breakLine());
+    }
+
+    // add data read from file to list
+    private static void addFileDataToList(String[] inputArgs) {
+        Task newTask;
+        String command = inputArgs[0];
+        boolean isMarked = Boolean.parseBoolean(inputArgs[1]);
+        switch (command) {
+        case "T":
+            newTask = new Todo(inputArgs[2]);
+            break;
+        case "D":
+            newTask = new Deadline(inputArgs[2], inputArgs[3]);
+            break;
+        case "E":
+            newTask = new Event(inputArgs[2], inputArgs[3], inputArgs[4]);
+            break;
+        default:
+            throw new IllegalStateException("(!) Invalid file contents.");
+        }
+        if (isMarked) {
+            newTask.mark();
+
+        }
+        tasks.add(newTask);
+    }
+
+    // add a new task
+    private static void addTodo(String taskInfo) {
+        Task newTask = new Todo(taskInfo);
+        tasks.add(newTask);
+        System.out.print(breakLine()
+                + "\tadded:\n\t\t" + newTask + '\n'
+                + "\t(total: " + tasks.size() + " tasks)\n"
+                + breakLine());
+    }
+
+    // create deadline
+    private static Deadline createDeadline(String taskInfo) {
+        String description, deadline;
+        String[] info = taskInfo.split("#by", 2);
+        description = info[0];
+        deadline = info[1];
+        return new Deadline(description, deadline);
+    }
+
+    // add a new deadline
+    private static void addDeadline(String taskInfo) {
+        Task newTask = createDeadline(taskInfo);
+        tasks.add(newTask);
+        System.out.print(breakLine()
+                + "\tadded:\n\t\t" + newTask + '\n'
+                + "\t(total: " + tasks.size() + " tasks)\n"
+                + breakLine());
+    }
+
+    // create new event
+    private static Event createEvent(String taskInfo) {
+        String description, from, to;
+        String[] info = taskInfo.split("#from", 2);
+        description = info[0];
+        info = info[1].split("#to", 2);
+        from = info[0];
+        to = info[1];
+        return new Event(description, from, to);
+    }
+
+    // add a new event
+    private static void addEvent(String taskInfo) {
+        Task newTask = createEvent(taskInfo);
+        tasks.add(newTask);
+        System.out.print(breakLine()
+                + "\tadded:\n\t\t" + newTask + '\n'
+                + "\t(total: " + tasks.size() + " tasks)\n"
+                + breakLine());
```

Comment:
This line seems similar to that in addTodo and addDeadline, perhaps it can be extracted into a separate method to enhance readability by maintaining a conistent level of abstraction. The same method call can be used across the other add... methods to reduce redundancy.